plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -165,6 +171,33 @@ static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMe
         return attrs.build();
     }
 
+    @SuppressForbidden(reason = "We call getInputStream in doPrivileged and provide SocketPermission")
+    private static Optional<String> getToken(String azMetadataTokenUrl) {
+        if (Strings.isNullOrEmpty(azMetadataTokenUrl)) {
+            return Optional.empty();
+        }
+        HttpURLConnection tokenUrlConnection;
+        try {
+            tokenUrlConnection = SocketAccess.doPrivilegedIOException(
+                () -> (HttpURLConnection) new URL(azMetadataTokenUrl).openConnection()
```

Comment:
I think we need to use a `PUT` request method here? https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

Reply:
You're absolutely right, it slipped from my mind!